### PR TITLE
minion/network:	Make Kelda's DNS server case insensitive.

### PR DESCRIPTION
**Pull request by Ch3ck**
* Kelda's DNS server should be case insensitive, converting hostnames to lower cases before atempting to resolve them in DNS.
* Fixes issue #1327

### CODE REVIEW

File: minion/network/dns.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ var table *dnsTable
 
 var dnsC = counter.New("Network Dns")
 
+var lookupHost = net.LookupHost
```
Comment (kklin) [inline minion/network/dns.go]: Moving this line of code isn't really relevant to the patch, and will change the Git history unnecessarily. Can you leave it as it was before?

Comment (kklin) [reply minion/network/dns.go]: Also, in the same vein of making the Git history cleaner, can you:
- Squash your three commits into a single commit. (`git log` is much easier to read if every commit is isolated, and it's easier to find a single commit that introduces a bug)
- This is really nitpicky, but I would make a couple small changes to your commit message for consistency with the rest of our messages. Again, this just makes `git log` a bit nicer to read:
    - Change "Made" to "Make". ([This](https://chris.beams.io/posts/git-commit/#imperative) post has a good explanation on why we prefer the imperative mood)
    - Remove the trailing period in the commit title (after "insensitive")
    - "converting hostnames to _lower cases_" should be "converting hostnames to _lowercase_"

File: minion/network/dns.go [hunk 2]
```diff
@@ -246,6 +249,7 @@ func hostnamesToDNS(hostnames []db.Hostname) map[string]net.IP {
 	records := map[string]net.IP{}
 	for _, hn := range hostnames {
 		if ip := net.ParseIP(hn.IP); ip != nil {
+			hn.Hostname = strings.ToLower(hn.Hostname)
```
Comment (kklin) [inline minion/network/dns.go]: Can you add a unit test for this PR? I think adding a case to the `TestLookupA` test would be sufficient

File: minion/network/dns.go (hunk 1 above)
Comment (Ch3ck) [reply minion/network/dns.go]: Let me make the necessary fixes.

File: minion/network/dns.go (hunk 2 above)
Comment (Ch3ck) [reply minion/network/dns.go]: Let me do that now.

**Comment by Ch3ck** [comment]
@kklin Issues fixed

File: integration-tester/tests/10-network/dns.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// This blueprint deploys a container that prints "Pass" if it successfully
```
Comment (kklin) [inline integration-tester/tests/10-network/dns.js]: Sorry, I meant adding a Go unit test here https://github.com/kelda/kelda/blob/master/minion/network/dns_test.go#L89. The integration tests require more setup, and I don't think it's worth it for this minor change.

Comment (Ch3ck) [reply integration-tester/tests/10-network/dns.js]: Yeah, that's true, let me do that real quick

Comment (Ch3ck) [reply integration-tester/tests/10-network/dns.js]: @kklin Done!

**Comment by kklin** [comment]
Thanks @Ch3ck!

**Comment by luise** [comment]
Thanks @Ch3ck !

**Comment by Ch3ck** [comment]
Onto the next one :-)

**Pull request merged**